packages/reaction-email-templates/server/hooks.js

```diff
@@ -0,0 +1,297 @@
+/**
+ * ReactionEmailFromTemplate - Returns a rendered and translated
+ * email template.
+ * @param {String} template name of the template
+ * @param {String} lng language of the email text
+ * @param {String} data data context for rendering
+ * @param {String} fallbackLng language that is used if lng has no translations
+ * @returns {Object} returns rendered HTML mail
+ */
+ReactionEmailFromTemplate = function (template, lng, data, fallbackLng) {
+  check(template, String);
+  check(lng, String);
+  check(data, Match.Optional(Object));
+  check(fallbackLng, Match.Optional(String));
+
+  if (typeof(fallbackLng) === "undefined") {
+    fallbackLng = "en";
+  } else if (fallbackLng !== "en") {
+    // also add english as second fallback..
+    fallbackLng = [fallbackLng, "en"];
+  }
+
+  if (!Template[template]) {
+      ReactionCore.Log.fatal(`Template '${ template }' does not exist. ` +
+        `Did you register it in the approriate workflow step (register.js)?`);
+  }
+  Template[template].helpers({
+    '_': function(key){
```

Comment:
_  for translations is quite common in python land. but you right: could be
confusing. please feel free to change anything you don't like or tell me
what to do

Sent from mobile device. Please excuse brevity and typos.
On Jun 6, 2016 10:28 PM, "Aaron Judd" <notifications@github.com> wrote:

> In packages/reaction-email-templates/server/hooks.js
> <https://github.com/reactioncommerce/reaction/pull/1062#discussion_r65962507>
> :
>
> > +  check(data, Match.Optional(Object));
> > +  check(fallbackLng, Match.Optional(String));
> > +
> > +  if (typeof(fallbackLng) === "undefined") {
> > +    fallbackLng = "en";
> > +  } else if (fallbackLng !== "en") {
> > +    // also add english as second fallback..
> > +    fallbackLng = [fallbackLng, "en"];
> > +  }
> > +
> > +  if (!Template[template]) {
> > +      ReactionCore.Log.fatal(`Template '${ template }' does not exist. ` +
> > +        `Did you register it in the approriate workflow step (register.js)?`);
> > +  }
> > +  Template[template].helpers({
> > +    '_': function(key){
>
> rather that we didn't have helpers/methods named _. Easily confused with
> underscore library, and also isn't descriptive. Hard to search for as well.
> This is one of the things I did not like about the tap-i18n package. This
> isn't doing the same thing the current i18n helper does?
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/reactioncommerce/reaction/pull/1062/files/2d32c98b38a87ca7cb62b2dccd52cc0c83af7f06#r65962507>,
> or mute the thread
> <https://github.com/notifications/unsubscribe/ABpybUmUYStJlBXXWItcvCmOU98S6TArks5qJIL5gaJpZM4IuPvR>
> .
>